module-1/lab-numpy/your-code/main.py

```diff
@@ -0,0 +1,197 @@
+#1. Import the NUMPY package under the name np.
+
+import numpy as np
+
+#2. Print the NUMPY version and the configuration.
+
+print (np.__version__)
+
+#3. Generate a 2x3x5 3-dimensional array with random values. Assign the array to variable "a"
+# Challenge: there are at least three easy ways that use numpy to generate random arrays. How many ways can you find?
+
+a = np.random.random((2,3,5))
+print (a)
+
+n=np.random.randint(1000000)
+print(n)
+a = np.random.randint(n, size=(2,3,5))
+print (a)
+
+
+n = np.random.random (100000)
+a=np.random.choice(n, size= (2,3,5))
+print (a)
+
+#4. Print a.
+
+a = np.random.random((2,3,5))
+print (a)
+
+
+n=np.random.randint(1000000)
+print(n)
+a = np.random.randint(n, size=(2,3,5))
+print (a)
+
+
+n = np.random.random (100000)
+a=np.random.choice(n, size= (2,3,5))
+
+print (a)
+print (a.shape)
+
+#5. Create a 5x2x3 3-dimensional array with all values equaling 1.
+#Assign the array to variable "b"
+
+b= np.ones ((5,2,3))
+
+
+#6. Print b.
+print (b)
+
+
+#7. Do a and b have the same size? How do you prove that in Python code?
+
+print(a.size)
+print (b.size)
+
+#8. Are you able to add a and b? Why or why not?
+
+# No, por que no tienen el mismo numero de filas columnas y grupos.
+# En el caso de que tuviesen la mismas dimensiones si que pondriamos
+
+
+#9. Transpose b so that it has the same structure of a (i.e. become a 2x3x5 array). Assign the transposed array to varialbe "c".
+
+c = np.transpose(b, (1, 2, 0))
+print (c)
+print(c.shape)
+
+#10. Try to add a and c. Now it should work. Assign the sum to varialbe "d". But why does it work now?
+
+d = np.add(a, c)
+print (d)
+
+ #Tienen las mismas dimensiones
+
+#11. Print a and d. Notice the difference and relation of the two array in terms of the values? Explain.
+
+print (a)
+print (d)
+
+#son exactamente iguales ya que multiplicas 1 por el numero aleatorio
```

Comment:
Which ones are equal? a and d are not equal , there is a difference between both that you must explain.